### PR TITLE
Ctl 003 create main UI

### PR DESCRIPTION
**Pull request by heupotato**
@lngyeen 
Pull Request: Task 3: Update UI for List of contact

### CODE REVIEW

File: lib/ui/contact-list.dart [hunk 1]
```diff
@@ -0,0 +1,96 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_contact_list/ui/contact-detail.dart';
```
Comment (lngyeen) [inline lib/ui/contact-list.dart]: File name should be 'contact_detail_screen.dart':
https://dart.dev/guides/language/effective-dart/style

File: lib/ui/main.dart [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_contact_list/ui/contact-list.dart';
```
Comment (lngyeen) [inline lib/ui/main.dart]: 1: File name should be 'contact_list_screen.dart':
https://dart.dev/guides/language/effective-dart/style
2: Should move the 'full screen' widgets to a sub folder 'screens' (/ui/screens)

File: lib/ui/contact-list.dart [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+        }
+        return Scaffold(
+            appBar: AppBar(
+                title: Text("Contact List"),
+                actions: [
+                    IconButton(icon: Icon(Icons.add_circle_outline_sharp), onPressed: _pushAdd),
+                ],
+            ),
+            body: Container(
```
Comment (lngyeen) [inline lib/ui/contact-list.dart]: Methods should not have more than an average of 30 code lines.
Should split into smaller (private) functions.

https://medium.com/swlh/coding-guidelines-for-dart-db22198b266b

File: lib/ui/contact-list.dart [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+class ContactList extends StatefulWidget {
+    const ContactList({ Key? key }) : super(key: key);
+
+    @override
+    _ContactListState createState() => _ContactListState();
+}
+
+class _ContactListState extends State<ContactList> { 
+    //create some fake data and assign to contactList 
+    List<Contact> contactList = List.generate(10, (int index) => Contact(id: index, firstName: "First ", lastName: "Last", phoneNumber: "0123456789"));
```
Comment (lngyeen) [inline lib/ui/contact-list.dart]: Should create another class called 'MockContact' put it in '/mock_data' folder, responsible for generating mock data. Example:
`class MockContact { 
static List<Contact> mocks() => ...
}`

File: pubspec.yaml [hunk 5]
```diff
@@ -33,6 +33,9 @@ dependencies:
   path_provider: ^2.0.2
   path: ^1.8.0
   moor_flutter: ^4.0.0
+  #UI
+  provider: ^3.0.0+1
```
Comment (lngyeen) [inline pubspec.yaml]: Avoid using 3rd lib. Please remove this.

File: lib/ui/contact-list.dart [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+                actions: [
+                    IconButton(icon: Icon(Icons.add_circle_outline_sharp), onPressed: _pushAdd),
+                ],
+            ),
+            body: Container(
+                padding: EdgeInsets.all(20), 
+                child: ListView.builder(
+                    itemCount: contactList.length,
+                    itemBuilder: (context, index){
+                        var contactName = contactList[index].firstName + " " + contactList[index].lastName;
```
Comment (lngyeen) [inline lib/ui/contact-list.dart]: avoid using var and dynamic:
https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-var-and-dynamic

**Review by lngyeen — CHANGES_REQUESTED** [review]

File: lib/ui/main.dart (hunk 2 above)
Comment (heupotato) [reply lib/ui/main.dart]: Changed the name of file for better style

**Comment by heupotato** [comment]
@lngyeen  I've resolved all of the change requests

File: lib/ui/screens/contact_list_screen.dart [hunk 7]
```diff
@@ -0,0 +1,90 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_contact_list/mockdata/mock_contact_list_data.dart';
```
Comment (lngyeen) [inline lib/ui/screens/contact_list_screen.dart]: file name should be mock_contact.dart.

**Pull request merged**